lib/client/conntest/connection_tester.go

```diff
@@ -41,6 +41,11 @@ type TestConnectionRequest struct {
 	ResourceKind string `json:"resource_kind"`
 	// ResourceName is the identification of the resource's instance to test.
 	ResourceName string `json:"resource_name"`
+	// ResourceTile is an optional field which identifies the Discover tile that wants to perform a
+	// connection test.
+	// Useful in situations in which the test for the same kind of resource needs to behave
+	// differently depending on the tile which sent the request.
+	ResourceTile string `json:"resource_tile"`
```

Comment:
I think we considered ResourceSubKind but that term is already in use by the Resource hierarchy, where, as you said, each resource can have multiple subkinds.
The issue is that this is not an actual subkind.
Server SubKinds are: teleport server, openssh or openssh-eice

In this case, we are talking about a Teleport Agent.
So the SubKind for this resource is already defined.
But we still need to differentiate it somehow.

We ended up with ResourceTile which has its downsides as well, but seemed to be our best bet (considering the other options).

Reply:
@marcoandredinis @kimlisa Given that there are conntest users which do not depend on Discover and that in the end this PR will add three new fields to `TestConnectionRequest`, what if I use `SSHNodeKind` instead of `ResourceTile`?

It's true that `Kind` is used for many other things, but here specifically we'd talk about `SSHNodeKind` which is specific to conntest.

The PR already needs to add `SSHPrincipalSelectionMode` and the next commit to address https://github.com/gravitational/teleport/pull/35308#discussion_r1414926618 will add `SSHNodeOS`.

Reply:
Ah, okay, from this angle `NodeSubKind` can indeed be confusing. `SSHNodeInstallationMode` sounds fine, though I'll shorten it to `SSHNodeSetupMethod` (`script`/`connect_my_computer`).

Reply:
Looks good :+1: 
Thank you for looking into this :pray: